src/skin/legacy/legacyskinparser.cpp

```diff
@@ -2306,7 +2307,9 @@ void LegacySkinParser::setupConnections(const QDomNode& node, WBaseWidget* pWidg
         ValueTransformer* pTransformer = nullptr;
         QDomElement transform = m_pContext->selectElement(con, "Transform");
         if (!transform.isNull()) {
-            pTransformer = ValueTransformer::parseFromXml(transform, *m_pContext);
+            pTransformer =
+                    ValueTransformer::parseFromXml(transform, *m_pContext)
+                            .get(); // TODO don't leak here
```

Comment:
`TODO`?  Could it be possible to make `pTransformer` a smart pointer?

Reply:
yes, but didn't want to here for the sake of keeping the scope small. Actually this is a bug, I want release here otherwise it'll be a double free...

Reply:
done